### DISCUSSION TITLE
Matplotlib in Jupyter Console?

### DISCUSSION THREAD

**Discussion by naquad** [jupyterlab/jupyterlab#16318] [Q&A]
Hi,

I'm looking to replace QtConsole with something capable of interactive plots while still being able to show the input & output caused by another client (think "Show All Console Activity"). Jupyter Lab's Console seems to be the thing, but I can't make it show plots. I've been trying different backends: widget, ipympl, and notebook, but none of them worked.

Am I missing something or its just impossible?

P. S. I've tried notebook but it is not capable of outputting another client's input/output.

**Comment by isatyamks** [comment]


**Reply by krassowski** [reply]
I do not see this comment as relevant to the discussion above. Did you mean to post it someplace else?

**Comment by krassowski** [comment]
@naquad

> Jupyter Lab's Console seems to be the thing, but I can't make it show plots.  I've been trying different backends: widget, ipympl, and notebook, but none of them worked.

It was able to show static plots so matplotlib's `inline` backend should have worked (but you do not mention it). For interactive plots the support for JupyterLab's Console was only added in ipywydgets [v8.1.3](https://github.com/jupyter-widgets/ipywidgets/releases/tag/8.1.3) which was just released last week.
